raiden/transfer/mediated_transfer/events.py

```diff
@@ -40,6 +43,11 @@ def balance_proof(self) -> BalanceProofUnsignedState:
         return self.transfer.balance_proof
 
 
+@dataclass(frozen=True)
+class RequestMetadata(Event):
+    dependant_events: List[SendMessageEvent]
```

Comment:
I know that for now it is only `SendMessageEvent` which can be wrapped. But it could be any type of event here, right?

Reply:
Yes, but mypy is opposing declaring this as a list of Events, as only SendMessageEvents are being used here right now. I decided to solve this when we have more types of events.

Reply:
hopefully it does not bite us in the future ;)

Reply:
It will for sure have to be changed, but it doesn't have to be changed now. I'd rather keep it like this for a first implementation.